### PR TITLE
Implement Issues Comments service

### PR DESCRIPTION
**Pull request by pdalpra**

### CODE REVIEW

File: src/main/scala/tv/teads/github/api/services/issues/IssuesCommentsService.scala [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+import tv.teads.github.api.http.mediatype.CommentBodyMediaType
+import tv.teads.github.api.http.mediatype.CommentBodyMediaType.Raw
+import tv.teads.github.api.model.issues.IssueComment
+import tv.teads.github.api.services.AbstractGithubService
+
+import scala.concurrent.Future
+
+class IssuesCommentsService(config: GithubApiClientConfig) extends AbstractGithubService(config) {
+
+  def listForIssue(repository: String, issueNumber: Long, mediaType: CommentBodyMediaType = Raw)(implicit ec: EC) =
```
Comment (pdalpra) [inline src/main/scala/tv/teads/github/api/services/issues/IssuesCommentsService.scala]: Codacy Issue found: [Public method must have explicit type](https://www.codacy.com/app/dalprapierre_2790/github-api-client_new/file/2574599639/issues/source?bid=3282900&fileBranchId=3298971#l13)

File: src/main/scala/tv/teads/github/api/services/issues/IssuesCommentsService.scala [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+      s"Could not fetch issue comments for issue $issueNumber on ${config.owner}/$repository"
+    )
+
+  def listForRepository(repository: String, mediaType: CommentBodyMediaType = Raw)(implicit ec: EC) =
+    json[List[IssueComment]](
+      getCall(s"repos/${config.owner}/$repository/issues/comments", mediaType.mediaType),
+      s"Could not fetch issue comments for ${config.owner}/$repository"
+    )
+
+  def get(repository: String, id: Long, mediaType: CommentBodyMediaType = Raw)(implicit ec: EC) =
```
Comment (pdalpra) [inline src/main/scala/tv/teads/github/api/services/issues/IssuesCommentsService.scala]: Codacy Issue found: [Public method must have explicit type](https://www.codacy.com/app/dalprapierre_2790/github-api-client_new/file/2574599639/issues/source?bid=3282900&fileBranchId=3298971#l25)

File: src/main/scala/tv/teads/github/api/services/issues/IssuesCommentsService.scala [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+
+class IssuesCommentsService(config: GithubApiClientConfig) extends AbstractGithubService(config) {
+
+  def listForIssue(repository: String, issueNumber: Long, mediaType: CommentBodyMediaType = Raw)(implicit ec: EC) =
+    json[List[IssueComment]](
+      getCall(s"repos/${config.owner}/$repository/issues/$issueNumber/comments", mediaType.mediaType),
+      s"Could not fetch issue comments for issue $issueNumber on ${config.owner}/$repository"
+    )
+
+  def listForRepository(repository: String, mediaType: CommentBodyMediaType = Raw)(implicit ec: EC) =
```
Comment (pdalpra) [inline src/main/scala/tv/teads/github/api/services/issues/IssuesCommentsService.scala]: Codacy Issue found: [Public method must have explicit type](https://www.codacy.com/app/dalprapierre_2790/github-api-client_new/file/2574599639/issues/source?bid=3282900&fileBranchId=3298971#l19)

**Comment by pdalpra** [comment]
Not sure why Codacy says there's code duplication when duplication patterns have been disabled -_-

File: src/test/scala/tv/teads/github/api/services/issues/IssuesCommentsServiceSpec.scala [hunk 4]
```diff
@@ -0,0 +1,165 @@
...
+  private val tests = List(
+    raw("**test comment**: *test message*"),
+    text("test comment: test message"),
+    html("<p><strong>test comment</strong>: <em>test message</em></p>")
+  )
+
+  "listForIssue" should {
+    "be able to fetch comments on 'Test issue' on github-api-client_test" in {
+      tests.foreach { test =>
+        whenReady(teadsClient.issues.comments.listForIssue("github-api-client_test", 1, test.mediaType)) { issues =>
```
Comment (gbougeard) [inline src/test/scala/tv/teads/github/api/services/issues/IssuesCommentsServiceSpec.scala]: maybe you might use a constant for `github-api-client_test` as it is used everywhere

Comment (pdalpra) [inline src/test/scala/tv/teads/github/api/services/issues/IssuesCommentsServiceSpec.scala]: Indeed, will refactor tests in the next PR

**Comment by pdalpra** [comment]
LGTY ?

**Comment by gbougeard** [comment]
LGTM

**Pull request merged**